src/audiomixerboard.cpp

```diff
@@ -1321,15 +1321,15 @@ void CAudioMixerBoard::AutoAdjustAllFaderLevels()
     QMutexLocker locker ( &Mutex );
 
     // initialize variables used for statistics
-    float vecMinLevel[MAX_NUM_FADER_GROUPS + 1];
     float vecMaxLevel[MAX_NUM_FADER_GROUPS + 1];
     int   vecChannelsPerGroup[MAX_NUM_FADER_GROUPS + 1];
     for ( int i = 0; i < MAX_NUM_FADER_GROUPS + 1; ++i )
     {
-        vecMinLevel[i] = UPPER_BOUND_SIG_METER;
         vecMaxLevel[i] = LOW_BOUND_SIG_METER;
         vecChannelsPerGroup[i] = 0;
     }
+    CVector<CVector<float>> levels;
```

Comment:
Can this be declared once in class scope, rather than each time `AutoAdjustAllFaderLevels` is called, as it does the `resize`?

In fact, all the variables that set up arrays might be better at class scope to get created just the once.  Would that affect things adversely?

Reply:
> Can this be declared once in class scope, rather than each time AutoAdjustAllFaderLevels is called, as it does the resize? 

Because of potential performance issues? Since this function is only called when the user *explicitly* calls it via the menu or keystroke, I would think that this should not be a problem. (In contrast to the audio processing functions which are called periodically and with a high frequency.)

> In fact, all the variables that set up arrays might be better at class scope to get created just the once. Would that affect things adversely?

Most of the other (fixed-sized) arrays are placed on the stack (not the heap) and therefore should not hurt performance.

> (Iterating over to clean each time is probably quicker than the memory allocation. I note there's no actual initialisation currently - is this method guaranteed to have zeroed values?)

Since only the outer `CVector` is resized, the STL should call the constructor of the inner `CVector` element, which should initialize with an empty list. New elements are added dynamically via `CVector::Add`.